### DISCUSSION TITLE
BYD B-box pro

### DISCUSSION THREAD

**Discussion by d4rkmen** [ai-republic/bms-to-inverter#57] [General]
I have this old battery and want to connect it to Deye sun inverter
And it's not supported offcourse. It has 4 BMS units connected by RS-485 to BMU, which has CAN connection to inverter

Interesting is there a protocol details for both BMS and BMU available?
And this current project supports some of those?

**Comment by d4rkmen** [comment]
![image](https://github.com/user-attachments/assets/7605c73c-8894-4a25-a155-662838a68793)

**Comment by ai-republic** [comment]
Have you tried the _BYD_CAN_ BMS binding?

**Reply by d4rkmen** [reply]
not have can transceiver yet, but ordered one already
but i have rs485 adapter and so far sniffing underlying bmu to bms protocol![image](https://github.com/user-attachments/assets/3eeacfa4-0ccf-4be0-b6ce-8c04dde5df01)

**Comment by ai-republic** [comment]
That looks good 😄 As I don't have the BYD RS485 protocol specification I'm not sure what it means. Let's wait for the CAN hat then we'll now more. PS: like the Halloween Pumkin 🤣
